auth/src/main/java/com/firebase/ui/auth/util/smartlock/SignInDelegate.java

```diff
@@ -0,0 +1,427 @@
+                                                   });
+
+        if (!madeAvailable
+                || !PlayServicesHelper.getInstance(getActivity()).isPlayServicesAvailable()
+                || !FirebaseAuthWrapperFactory.getFirebaseAuthWrapper(mHelper.getAppName())
+                .isPlayServicesAvailable(getActivity())) {
+            Log.w(TAG, "playServices: could not make available.");
+            finish(RESULT_CANCELED, new Intent());
+            return;
+        }
+
+        if (mHelper.getFlowParams().smartLockEnabled) {
+            mHelper.showLoadingDialog(R.string.progress_dialog_loading);
+            initGoogleApiClient(null);
+            mHelper.getCredentialsApi()
+                    .request(mGoogleApiClient,
+                             new CredentialRequest.Builder()
+                                     .setPasswordLoginSupported(true)
+                                     .setAccountTypes(IdentityProviders.GOOGLE,
+                                                      IdentityProviders.FACEBOOK,
+                                                      IdentityProviders.TWITTER)
+                                     .build())
+                    .setResultCallback(this);
+        } else {
+            startAuthMethodChoice();
+        }
+    }
+
+    @Override
+    public void onResult(@NonNull CredentialRequestResult result) {
+        Status status = result.getStatus();
+
+        if (status.isSuccess()) {
+            // Auto sign-in success
+            handleCredential(result.getCredential());
+            String email = getEmailFromCredential();
+            String password = getPasswordFromCredential();
+
+            if (TextUtils.isEmpty(password)) {
+                // log in with id/provider
+                redirectToIdpSignIn(email, getAccountTypeFromCredential());
+            } else {
+                // Sign in with the email/password retrieved from SmartLock
+                signInWithEmailAndPassword(email, password);
+            }
+        } else if (status.getStatusCode() == CommonStatusCodes.RESOLUTION_REQUIRED) {
+            // resolve saved emails
+            try {
+                startIntentSenderForResult(status.getResolution().getIntentSender(),
+                                           RC_CREDENTIALS_READ,
+                                           null,
+                                           0,
+                                           0,
+                                           0,
+                                           null);
+            } catch (IntentSender.SendIntentException e) {
+                Log.e(TAG, "Failed to send Credentials intent.", e);
+            }
+        }
+    }
+
+    @Override
+    public void onConnected(@Nullable Bundle bundle) {
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (BuildConfig.DEBUG) {
```

Comment:
I just copied most of this stuff from `ChooseAccountActivity` so I'll just remove this for now, but if you want to keep it just tell me and I'll add it back.